pkgs/build-support/php/build-php.nix

```
@@ -0,0 +1,102 @@
+{ stdenvNoCC, lib, writeTextDir, composer, php, unzip, xz }@inputs:
+
+{ pname
+, version
+, src
+, composerLock ? "null"
+, vendorDir ? "vendor"
+, vendorHash
+, php ? inputs.php
+, composer ? inputs.composer
+, ...
+}@args:
+
+let
+  vendor = stdenvNoCC.mkDerivation (finalAttrs: {
+    inherit src version;
+
+    pname = "${pname}-vendor";
+
+    nativeBuildInputs = [
+      composer
+      unzip
+      xz
+    ];
+
+    strictDeps = true;
+
+    buildPhase = let
+      # See https://github.com/composer/composer/pull/9212
+      prefix = "${builtins.replaceStrings ["-"] [""] pname}";
+
+      # See https://getcomposer.org/doc/03-cli.md#composer-home-config-json
+      composerHome = let
+        content = {
+          config = {
+            autoloader-suffix = prefix;
+          };
+        };
+      in
+        writeTextDir "/config.json" (builtins.toJSON content);
+    in ''
+      runHook preBuild
+
+      COMPOSER_HOME=${composerHome} \
+      COMPOSER_VENDOR_DIR=${vendorDir} \
+
+      if [ ! -f "${src}/composer.lock" ] && [ ! -z "${composerLock}" ] && [ -f "${composerLock}" ]
```

The first check would disallow overriding a lock existing in a source and the third would silently ignore broken 
`composerLock`

```suggestion
      if [[ ! -z "${composerLock}" ]]
```

---

I made something a little bit different, let me know if it's ok

---

That suffers from the same issues, only the syntax changed a bit.